src/commands/scaling/addScaleRule/GetScaleRuleNameStep.ts

```diff
@@ -0,0 +1,54 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ContainerApp, ContainerAppsAPIClient, ScaleRule } from '@azure/arm-appcontainers';
+import { AzureWizardPromptStep } from '@microsoft/vscode-azext-utils';
+import { createContainerAppsAPIClient } from '../../../utils/azureClients';
+import { getResourceGroupFromId } from '../../../utils/azureUtils';
+import { localize } from '../../../utils/localize';
+import { IAddScaleRuleWizardContext } from './IAddScaleRuleWizardContext';
+
+export class GetScaleRuleNameStep extends AzureWizardPromptStep<IAddScaleRuleWizardContext> {
+    public hideStepCount: boolean = true;
+    containerApp: ContainerApp | undefined;
+    scaleRules: ScaleRule[] | undefined;
+
+    public async prompt(context: IAddScaleRuleWizardContext): Promise<void> {
+        context.ruleName = (await context.ui.showInputBox({
+            prompt: localize('scaleRuleNamePrompt', 'Enter a name for the new scale rule.'),
+            validateInput: async (value: string | undefined): Promise<string | undefined> => await this.validateInput(context, value)
+        })).trim();
+    }
+
+    public shouldPrompt(): boolean {
+        return true;
```

Comment:
Typically, we check the value to see if it should prompt or not. Something like `return context.ruleName !== undefined`. The reason being, if we ever make a "basic" version of this command and set the `ruleName` to some default value without user input, the prompt step would be skipped.

Reply:
I'd do `return !context.ruleName`, so any falsy value will prompt.

Reply:
Ah sorry, yeah, `context.ruleName === undefined` is what I meant. I tend to be specific about this incase a "" is a valid value or if the step deals with boolean values, but generally `!context.ruleName` works as well.

Reply:
It's all good, I already understood what you meant :)

Reply:
Yeah, I try to be deliberate in my thinking about strings w.r.t. `undefined` vs `''` vs `null` etc. _Most_ of the time (but certainly not _all_ of the time), any falsy value is semantically the same. Fortunately `null` rarely ever shows up (though argh the `undefined` vs `null` thing is a nightmare). It's up to you to decide if `''` is an acceptable value but for a resource name I would think it is not.

I also generally avoid doing equality checks against `undefined` _because_ of that `null` vs. `undefined` mess, unless I know with certainty that `undefined` is the thing I'm looking for.

Reply:
In our case here, would it make sense to do an input validation check instead then?  Maybe something like this

`
public shouldPrompt(context): boolean {
        return this.validateInput(context.ruleName);
}
`

Reply:
Would it make sense to leverage our `validateInput` function to help us here then?  Maybe something like...

`
function shouldPrompt(context): boolean {
  return this.validateInput(context.ruleName) !== undefined;
}
`

Reply:
That's not really how `shouldPrompt` is used. It gets called when the wizard is created to get the total number of steps. If anything returns false at that time, it will not be included in total steps. It is then called again _before_ prompt is run, to see if it should skip it or not. 

If you tried to use validate input in there, there'd be no context.ruleName at the time of creation.